File: src/kyu_6/street_fighter_2_character_selection/python/street_fighter_2_character_selection.py
def street_fighter_selection(fighters, initial_position, moves):
    cur_pos = [initial_position[0], initial_position[1]]
    selected_fighters = []

    for move in moves:
        if move == "up":
            cur_pos[0] = 0
        elif move == "down":
            cur_pos[0] = 1
        elif move == "right":
            cur_pos[1] = (cur_pos[1] + 1) % 6
        elif move == "left":
            cur_pos[1] = (cur_pos[1] - 1) % 6
        selected_fighters.append(fighters[cur_pos[0]][cur_pos[1]])

    return selected_fighters

File: src/kyu_6/street_fighter_2_character_selection/csharp/street_fighter_2_character_selection.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Kata
{
    public string[] StreetFighterSelection(string[][] fighters, int[] position, string[] moves)
    {
    string[] solution = new string[moves.Length];
    int row = position[0]; int col = position[1];
    for(int i = 0; i < moves.Length; i++){
        switch(moves[i]){
            case "up" :
                row = 0;
                break;
            case "down" :
                row = 1;
                break;
            case "left" :
                col = col == 0 ? 5 : col - 1;
                break;
            case "right" :
                col = col == 5 ? 0 : col + 1;
                break;
        }
        solution[i] = fighters[row][col];
    }
    return solution;
    }
}